### PR TITLE
Add `wide` breakpoint and `useResponsiveValue`

### PR DESCRIPTION
**Pull request by markdalgleish**
This also deprecates `useBreakpoint`.

See changesets for more details.

### CODE REVIEW

File: .changeset/cyan-points-happen.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ - The `breakpoints` object, which now exposes the number `1200` via `breakpoints.wide`, i.e.
+   ```ts
+   {
+     mobile: 0,
+     tablet: 740,
+     desktop: 940,
+     wide: 1200
+   }
+   ```
+ - The `useBreakpoint` Hook, which now returns `"wide"` on screens from 1200px and above.
```
Comment (askoufis) [inline .changeset/cyan-points-happen.md:50]: Doesn't this contradict [this statement](https://github.com/seek-oss/braid-design-system/pull/960/files#diff-71b58a179f50e429ac88569d49b52824e09495c71a8a563d8b708f6cd1f293ebR22)?

Comment (markdalgleish) [reply .changeset/cyan-points-happen.md:50]: Oops, nice catch 👍

File: .changeset/blue-toes-kiss.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Deprecate `useBreakpoint`
+
+This Hook has been deprecated in favour of the new `useResponsiveValue` Hook.
+
+This is because `useBreakpoint` leads consumers to
+inadvertently couple themselves to the current set of breakpoints,
+making it risky for us to introduce new breakpoints.
+
+For example, you
+may have chosen to detect large screens by checking that the user is on the
```
Comment (michaeltaranto) [inline .changeset/blue-toes-kiss.md:19]: The formatting here seem really odd?

File: .changeset/cyan-points-happen.md (hunk 1 above)
Comment (markdalgleish) [reply .changeset/cyan-points-happen.md:50]: Writing this is actually what made me realise it was going to be a problem.

File: lib/components/useBreakpoint/useBreakpoint.docs.tsx [hunk 3]
```diff
@@ -29,10 +49,11 @@ const docs: ComponentDocs = {
       description: (
         <Text>
           This Hook will return the breakpoint the browser viewport currently
-          falls within (<Strong>mobile</Strong>, <Strong>tablet</Strong> or{' '}
-          <Strong>desktop</Strong>). As this can only be calculated in the
-          browser, the value will also be <Strong>null</Strong> when rendering
-          server-side or statically rendering. Window resizing is supported.
+          falls within (<Strong>mobile</Strong>, <Strong>tablet</Strong>,{' '}
+          <Strong>desktop</Strong> or <Strong>wide</Strong>). As this can only
```
Comment (michaeltaranto) [inline lib/components/useBreakpoint/useBreakpoint.docs.tsx:53]: We don't return `wide` from this Hook anymore.

**Review by michaeltaranto — APPROVED** [review]

**Pull request merged**